### PR TITLE
2022-01-12 RLS-2899  GUARD-2312 ChannelAdvisor session mark and cancellation token refactoring

### PR DESCRIPTION
**Pull request by maxim-saltanov-skuvault**
**fixes / improvements**
- changed order of the method parameters - I've set CancellationToken as latest parameter across methods as Nathan suggested. Also I've made it optional as it is not required for SOAP (never used) but it uses for REST. 
- made SessionMark as mandatory parameter across methods
- added some missed logz and errors handling across some methods

**commits**: https://github.com/skuvault-integrations/channelAdvisorAccess/commit/98d3e86f630046c236c07a7670332a73c25f2fed and https://github.com/skuvault-integrations/channelAdvisorAccess/commit/801911a8246af59d94c05f40a5a4befad94e8cb5 

**PS**. This branch also contains the changes from other channel advisor tickets https://agileharbor.atlassian.net/browse/GUARD-2225, https://agileharbor.atlassian.net/browse/GUARD-2226 and https://agileharbor.atlassian.net/browse/GUARD-2307

### CODE REVIEW

File: src/ChannelAdvisorAccess/REST/Services/Orders/OrdersService.cs [hunk 1]
```diff
@@ -150,13 +150,10 @@ public DateTime LastActivityTime
 		/// <param name="filter"></param>
 		/// <param name="mark"></param>
 		/// <returns></returns>
-		public async Task< IEnumerable< T > > GetOrdersAsync < T >( string filter, CancellationToken token, Mark mark, int? operationTimeout = null )
+		public async Task< IEnumerable< T > > GetOrdersAsync < T >( string filter, Mark mark, int? operationTimeout = null, CancellationToken token = default )
```
Comment (maxim-skuvault) [inline src/ChannelAdvisorAccess/REST/Services/Orders/OrdersService.cs:153]: @maxim-saltanov-skuvault , Is this branch building with release.cmd? I wonder if `default` is a recent language feature that won't compile with the [version of MSBuild we're using](https://github.com/skuvault-integrations/channelAdvisorAccess/blob/master/.build.ps1#L25)

**Comment by maxim-skuvault** [comment]
@maxim-saltanov-skuvault , See inline comment and also a comment on this ticket in JIRA

File: src/ChannelAdvisorAccess/REST/Services/Orders/OrdersService.cs (hunk 1 above)
Comment (maxim-saltanov-skuvault) [reply src/ChannelAdvisorAccess/REST/Services/Orders/OrdersService.cs]: @maxim-skuvault thanks Max, you're right. I've fixed it and it successfully compile now

**Review by maxim-skuvault — APPROVED** [review]

**Review by nathan-skuvault — APPROVED** [review]
so many marks!

**Review by maxim-skuvault — CHANGES_REQUESTED** [review]
@maxim-saltanov-skuvault , The cancellation token and mark  changes look good. However, just like the earlier PR (2307, I think) this shows RedAlert's changes (Netco, Polly, etc.) even though those changes are already in master - [example](https://github.com/skuvault-integrations/channelAdvisorAccess/blob/master/src/ChannelAdvisorAccess/ChannelAdvisorAccess.csproj#L138). Please try retargeting the PR to another branch and then back to master, to get the diffs to only show your changes.

File: src/ChannelAdvisorAccess/Service References/AdminService/ChannelAdvisorAccess.AdminService.PingResponse.datasource [hunk 2]
```diff
@@ -6,5 +6,5 @@
     cause the file to be unrecognizable by the program.
 -->
 <GenericObjectDataSource DisplayName="PingResponse" Version="1.0" xmlns="urn:schemas-microsoft-com:xml-msdatasource">
-   <TypeInfo>ChannelAdvisorAccess.AdminService.PingResponse, Service References.AdminService.Reference.cs, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null</TypeInfo>
+   <TypeInfo>ChannelAdvisorAccess.AdminService.PingResponse, Service References.AdminService.Reference.cs.dll, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null</TypeInfo>
```
Comment (nathan-newman65190) [inline src/ChannelAdvisorAccess/Service References/AdminService/ChannelAdvisorAccess.AdminService.PingResponse.datasource:9]: What are these changes for?

File: src/ChannelAdvisorAccess/Service References/FulfillmentService/Reference.cs [hunk 3]
```diff
@@ -57,7 +57,7 @@ public interface FulfillmentServiceSoap {
     }
     
     /// <remarks/>
-    [System.CodeDom.Compiler.GeneratedCodeAttribute("System.Xml", "4.7.2102.0")]
+    [System.CodeDom.Compiler.GeneratedCodeAttribute("System.Xml", "4.8.4084.0")]
```
Comment (nathan-newman65190) [inline src/ChannelAdvisorAccess/Service References/FulfillmentService/Reference.cs:60]: Why the update to these?

File: src/ChannelAdvisorAccess/Services/Items/ServiceBaseAbstr.cs [hunk 4]
```diff
@@ -44,26 +44,12 @@ protected string CreateMethodCallInfo( string methodParameters = "", string payl
...
-			clientChannel.Closing += TrackSoapClientNetworkActivity;
-			clientChannel.Closed += TrackSoapClientNetworkActivity;
-			clientChannel.Faulted += TrackSoapClientNetworkActivity;
-		}
-
-		private void TrackSoapClientNetworkActivity( object sender, EventArgs args )
-		{
-			RefreshLastNetworkActivityTime();
-		}
-
```
Comment (nathan-newman65190) [inline src/ChannelAdvisorAccess/Services/Items/ServiceBaseAbstr.cs:61]: Is this not needed for anything?

File: src/ChannelAdvisorAccess/app.config [hunk 5]
```diff
@@ -42,14 +41,18 @@
 		</client>
 	</system.serviceModel>
 	<startup>
-		<supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.1" />
+		<supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.8" />
```
Comment (nathan-newman65190) [inline src/ChannelAdvisorAccess/app.config:44]: Was this intentional?

**Review by nathan-newman65190 — CHANGES_REQUESTED** [review]
Mark stuff looks good, but it looks like a lot of files are being changed that have nothing to do with the mark refactor.

**Comment by maxim-saltanov-skuvault** [comment]
> @maxim-saltanov-skuvault , The cancellation token and mark changes look good. However, just like the earlier PR (2307, I think) this shows RedAlert's changes (Netco, Polly, etc.) even though those changes are already in master - [example](https://github.com/skuvault-integrations/channelAdvisorAccess/blob/master/src/ChannelAdvisorAccess/ChannelAdvisorAccess.csproj#L138). Please try retargeting the PR to another branch and then back to master, to get the diffs to only show your changes.

> Please try retargeting the PR to another branch and then back to master, to get the diffs to only show your changes.

Thanks @maxim-skuvault , it helped, I've fixed, and it seems all below's comments outdated now

**Review by chris-skuvault — APPROVED** [review]

**Review by maxim-skuvault — APPROVED** [review]
@maxim-saltanov-skuvault , This looks good presuming that the methods that got moved (UpdateOrderListAsync, GetOrdersAsync) didn't have any changes besides CancellationToken & mark

**Pull request merged**